utils/firebaseNotification.js

```diff
@@ -0,0 +1,74 @@
+import messaging from '@react-native-firebase/messaging'
+import { Alert } from 'react-native';
+
+
+export default class FirebaseNotification{
+
+async checkPermission(){
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+    if (enabled) {
+      this.getFcmToken() 
+      console.log('Authorization status:', authStatus);
+    }
+}
+
+async getFcmToken(){
+    const fcmToken = await messaging().getToken();
+    if (fcmToken) {
+     console.log("Your Firebase Token is:", fcmToken);
+    } else {
+     console.log("Failed", "No token received");
+    }
+  }
+
+  foregroundMessage(){
+    const unsubscribe = messaging().onMessage(async remoteMessage => {
+      console.log("Foreground is running")
+        this.handleNotificationEvent(remoteMessage.notification.title,remoteMessage.notification.body)
+      });
+      return unsubscribe;
+  }
+
+  backgroundMessageHandler(){
+    messaging().setBackgroundMessageHandler(async remoteMessage => {
+      console.log("Background is running")
+      this.handleNotificationEvent(remoteMessage.notification.title,remoteMessage.notification.body)
```

Comment:
this line should be refactored using destructuring as well

Reply:
fixed